src/v/serde/test/serde_test.cc

```diff
@@ -829,3 +829,45 @@ SEASTAR_THREAD_TEST_CASE(type_str) {
       serde::type_str<std::string>().find("string"), std::string_view::npos);
     BOOST_CHECK_EQUAL(serde::type_str<foo_bar>(), "foo_bar");
 }
+// Utility to serialize and deserialize back an input.
+template<class T>
+T serde_input(T input) {
+    return serde::from_iobuf<T>(serde::to_iobuf(std::move(input)));
+}
+
+SEASTAR_THREAD_TEST_CASE(duration_type_test) {
+    using namespace std::chrono;
+    constexpr hours h(1);
+    constexpr milliseconds ms{3};
+    constexpr milliseconds ms_neg{-1};
+    constexpr duration<int, std::kilo> ks(3);
+    constexpr duration<int16_t> int16t_d(10);
+    constexpr duration<int16_t> int32t_d(10);
```

Comment:
Also `duration<int64_t,std::pico>` to check paths with `ratio::den!=1`?

Reply:
Good idea.. done.. we need to be careful there as there is loss of precision converting to nanoseconds :laughing:

Reply:
nice idea.